Sources/Networking/HTTPClient/HTTPResponse.swift

```diff
@@ -18,19 +18,54 @@ struct HTTPResponse<Body: HTTPResponseBody> {
 
     typealias Result = Swift.Result<Self, NetworkError>
 
-    let statusCode: HTTPStatusCode
-    let body: Body
+    var statusCode: HTTPStatusCode
+    var responseHeaders: HTTPClient.ResponseHeaders
+    var body: Body
+    var validationResult: HTTPResponseValidationResult
```

Comment:
I may be missing something but why did we make these `var` instead of `let`?

Reply:
So we can modify the validation result after it's initially created in `HTTPClient` (just an implementation detail).
Note that `struct`s have [value semantics](https://www.swiftbysundell.com/articles/utilizing-value-semantics-in-swift/), so this isn't like having mutable data in other languages.

Example:
```swift
struct Data {
  // Technically mutable
  var a: Int
  var b: String
}

let d = Data(a: 1, b: "test")
d.a +=1 // Error, `d` is immutable!

func f(_ d: Data) {
  d.a += 1 // Error, d is immutable
}
```

Additionally, `struct`s get implicitly copied (they have _value_ semantics, not _reference_), they're passed by _value_, so you can't lead to shared mutable state.

So basically it doesn't make a difference if these are `var`s, it just makes creating copies with different values easier without needing to do it manually calling the constructor again.

Reply:
Right, I asked about this before and I forgot 😅 thanks for explaining again!